george69/Interpreter.py

```diff
@@ -0,0 +1,389 @@
+from Instruction import Instruction
+from Datum import Datum
+import struct
+import sys
+import filecmp
+class Interpreter(object):
+    def __init__(self):
+        self.rstack = [] #list of Datum
+        self.sp = -1
+        self.fpstack = []
+        self.fpsp = -1
+        self.pc = 0
+        self.mem = [] ##list of bytes
+        self.availableInstructions = {
+            "cmpe": self.cmpe,
+            "cmplt": self.cmplt,
+            "cmpgt": self.cmpgt,
+            "jmp": self.jmp,
+            "jmpc": self.jmpc,
+            "call": self.call,
+            "ret": self.ret,
+            "pushc": self.pushc,
+            "pushs": self.pushs,
+            "pushi": self.pushi,
+            "pushf": self.pushf,
+            "pushvc": self.pushvc,
+            "pushvs": self.pushvs,
+            "pushvi": self.pushvi,
+            "pushvf": self.pushvf,
+            "popm": self.popm,
+            "popa": self.popa,
+            "popv": self.popv,
+            "peekc": self.peekc,
+            "peeks": self.peeks,
+            "peeki": self.peeki,
+            "peekf": self.peekf,
+            "pokec": self.pokec,
+            "pokes": self.pokes,
+            "pokei": self.pokei,
+            "pokef": self.pokef,
+            "swp": self.swp,
+            "add": self.add,
+            "sub": self.sub,
+            "mul": self.mul,
+            "div": self.div,
+            "printc": self.printc,
+            "prints": self.prints,
+            "printi": self.printi,
+            "printf": self.printf,
+            "halt": self.halt
+        }
+
+    def loadProgram(self, fileName):
+        with open(fileName, "rb") as fp:
+            self.mem = bytearray(fp.read())
+    def executeProgram(self):
+        try:
+            while True:
+                instruction = Instruction(self.mem[self.pc])
+                ##print(instruction.numericalInstruction)
+                #print(instruction.instruction)
+                #print(self.pc)
+                fun = self.availableInstructions.get(instruction.instruction)
+                fun()
+                #print([str(d) for d in self.rstack])
+                if instruction.instruction == "halt":
+                    break
+        except Exception as e:
+            print("ERROR OCCURED, DUMPING")
+            print(e)
+            self.halt()
+
+    def halt(self):
+        print()
+        print("Compile values:")
+        print(str(self), end="")
+
+    def call(self):
```

Comment:
It could be helpful if each function had a comment describing what it is supposed to achieve.